api/client/client.go

```diff
@@ -233,49 +235,51 @@ func (srv *server) authReq(req, ID []byte, provider auth.Provider, target string
 
 	jsonData, err := json.Marshal(aReq)
 	if err != nil {
-		return nil, errors.Wrap(errors.APIClientError, errors.JSONError, err)
+		return nil, nil, errors.Wrap(errors.APIClientError, errors.JSONError, err)
 	}
 
 	response, err := srv.post(url, jsonData)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
 	result, ok := response.Result.(map[string]interface{})
 	if !ok {
-		return nil, errors.New(errors.APIClientError, errors.JSONError)
+		return nil, nil, errors.New(errors.APIClientError, errors.JSONError)
 	}
 
-	var cert string
+	var ca, cert string
 	if returnBundle {
 		bundle, okBundle := result["bundle"].(map[string]interface{})
 		if !okBundle {
-			return nil, errors.New(errors.APIClientError, errors.JSONError)
+			return nil, nil, errors.New(errors.APIClientError, errors.JSONError)
 		}
 		cert, ok = bundle["bundle"].(string)
+		ca, _ = bundle["root"].(string)
```

Comment:
Is it okay to discard `ok` here and assume the "root" entry is present?

Reply:
I was not exactly sure but I did not want to return an error in that case because returning just the bundle is still valid/useful. The API docs (https://github.com/cloudflare/cfssl/blob/master/doc/api/endpoint_bundle.txt) unfortunately are not exactly clear about if the response is guaranteed to contain the root entry.